lottie/src/main/java/com/airbnb/lottie/animation/content/ContentGroup.java

```diff
@@ -153,21 +160,48 @@ Matrix getTransformationMatrix() {
       return;
     }
     matrix.set(parentMatrix);
-    int alpha;
+    int layerAlpha;
     if (transformAnimation != null) {
       matrix.preConcat(transformAnimation.getMatrix());
       int opacity = transformAnimation.getOpacity() == null ? 100 : transformAnimation.getOpacity().getValue();
-      alpha = (int) ((opacity / 100f * parentAlpha / 255f) * 255);
+      layerAlpha = (int) ((opacity / 100f * parentAlpha / 255f) * 255);
     } else {
-      alpha = parentAlpha;
+      layerAlpha = parentAlpha;
     }
 
+    // Apply off-screen rendering only when needed in order to improve rendering performance.
+    boolean isRenderingWithOffScreen = lottieDrawable.isApplyingOpacityToLayersEnabled() && hasTwoOrMoreDrawableContent() && layerAlpha != 255;
+    if (isRenderingWithOffScreen) {
+      offScreenRectF.set(0, 0, 0, 0);
+      getBounds(offScreenRectF, matrix, true);
+      offScreenPaint.setAlpha(layerAlpha);
+      Utils.saveLayerCompat(canvas, offScreenRectF, offScreenPaint);
+    }
+
+    int childAlpha = isRenderingWithOffScreen ? 255 : layerAlpha;
     for (int i = contents.size() - 1; i >= 0; i--) {
       Object content = contents.get(i);
       if (content instanceof DrawingContent) {
-        ((DrawingContent) content).draw(canvas, matrix, alpha);
+        ((DrawingContent) content).draw(canvas, matrix, childAlpha);
+      }
+    }
+
+    if (isRenderingWithOffScreen) {
+      canvas.restore();
+    }
+  }
+
+  private boolean hasTwoOrMoreDrawableContent() {
```

Comment:
@r-ralph This function name isn't consistent with what it's doing.
However, it does seem smart to not do off screen rendering if there is only one drawing content. That's just not what this function is doing.

Reply:
Sorry, my understanding might be wrong.
This function only checks the list `contents`, and I think it is consistent with the name. 
It is the caller's responsibility to decide whether to do off-screen drawing with this information, so I think it should not be changed to a name related to offscreen.

Reply:
@r-ralph Ah, I see how this function works. It's just a bit confusing the way it's written because it is called `hasTwoOrMoreDrawableContent` but has a variable `hasOneOrMoreDrawableContent` that gets set once and then returns true the next iteration with `DrawingContent`. Looks like it would work but it's confusing to read. What if you just had an int counter, increment it at the beginning of the instanceof block and then return right after if it's `>= 2`. I think that would be easier to read. What do you think?

Reply:
I got it! I'll address it.